src/chessboard/components/SparePiece.tsx

```diff
@@ -0,0 +1,59 @@
+import { ReactNode } from "react";
+import { useDrag } from "react-dnd";
+import { getEmptyImage } from "react-dnd-html5-backend";
+
+import { CustomPieceFn, Piece as Pc } from "../types";
+import { defaultPieces } from "../media/pieces";
+
+type PieceProps = {
+  piece: Pc;
+  width: number;
+  customPieceJSX?: CustomPieceFn;
+  dndId: string;
+};
+
+export const SparePiece = ({
+  piece,
+  width,
+  customPieceJSX,
+  dndId,
+}: PieceProps) => {
+  const renderPiece = customPieceJSX ?? defaultPieces[piece];
+  const [{ canDrag, isDragging }, drag, dragPreview] = useDrag(
+    () => ({
+      type: "piece",
+      item: () => {
+        return { piece, isSpare: true, id: dndId };
```

Comment:
I think according to the handleDrop function below, `square` is a required prop that isn't passed here

Reply:
I've made typescript fix, now it will be cleaner